syz-manager/html.go

```diff
@@ -208,26 +210,42 @@ func (mgr *Manager) httpCorpus(w http.ResponseWriter, r *http.Request) {
 	executeTemplate(w, corpusTemplate, data)
 }
 
+const (
+	DoHTML int = iota
+	DoHTMLTable
+	DoCSV
+	DoCSVFiles
+)
+
 func (mgr *Manager) httpCover(w http.ResponseWriter, r *http.Request) {
+	mgr.httpCoverCover(w, r, DoHTML, true)
+}
+
+func (mgr *Manager) httpSubsystemCover(w http.ResponseWriter, r *http.Request) {
+	mgr.httpCoverCover(w, r, DoHTMLTable, true)
+}
+
+func (mgr *Manager) httpCoverCover(w http.ResponseWriter, r *http.Request, funcFlag int, isHTMLCover bool) {
 	if !mgr.cfg.Cover {
-		mgr.mu.Lock()
-		defer mgr.mu.Unlock()
-		mgr.httpCoverFallback(w, r)
+		if isHTMLCover {
+			mgr.mu.Lock()
+			defer mgr.mu.Unlock()
+			mgr.httpCoverFallback(w, r)
+		} else {
+			http.Error(w, "coverage is not enabled", http.StatusInternalServerError)
+			return
```

Comment:
Oh, I think we actually need this return in both cases. I see this is an old bug.
Please move the return one below.